src/main/java/org/cactoos/io/Zip.java

```diff
@@ -48,14 +48,23 @@ public final class Zip implements Input {
     /**
      * Origin directory.
      */
-    private final Directory origin;
+    private final Iterable<? extends Path> origin;
 
     /**
      * Ctor.
      *
      * @param origin Origin directory.
      */
     public Zip(final Directory origin) {
+        this((Iterable<? extends Path>) origin);
```

Comment:
@DmitryBarskov why is this cast needed? I'm surprised this just doesn't work without it...

Reply:
@victornoel I did it to avoid recursive invocation since in this case `Zip(Directory)` is invoked.
In fact this ctor is not needed as any `Directory` instance is a `Iterable<Path>` already and `Zip(Iterable<Path>)` can be called `Directori`es already. I just thought having `Zip(Directory)` is good for readability or something like that.

Reply:
@DmitryBarskov right! in that case let get rid of this constructor :) can you can mention `Directory` in the javadoc as an example of using `Zip`?